packages/flutter/test/material/snack_bar_theme_test.dart

```diff
@@ -0,0 +1,156 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  test('SnackBarThemeData copyWith, ==, hashCode basics', () {
+    expect(const SnackBarThemeData(), const SnackBarThemeData().copyWith());
+    expect(const SnackBarThemeData().hashCode, const SnackBarThemeData().copyWith().hashCode);
+  });
+
+  testWidgets('Passing no SnackBarThemeData returns defaults', (WidgetTester tester) async {
+    await tester.pumpWidget(MaterialApp(
+      home: Scaffold(
+        body: Builder(
+          builder: (BuildContext context) {
+            return GestureDetector(
+              onTap: () {
+                Scaffold.of(context).showSnackBar(SnackBar(
+                  content: const Text('I am a snack bar.'),
+                  duration: const Duration(seconds: 2),
+                  action: SnackBarAction(label: 'ACTION', onPressed: () {}),
+                ));
+              },
+              child: const Text('X'),
+            );
+          }
+        ),
+      ),
+    ));
+
+    await tester.tap(find.text('X'));
+    await tester.pump(); // start animation
+    await tester.pump(const Duration(milliseconds: 750));
+
+    final Material material = _getSnackBarMaterial(tester);
+
+    expect(material.color, const Color(0xFF323232));
+    expect(material.elevation, 6.0);
+    expect(material.shape, null);
+  });
+
+  testWidgets('SnackBar uses values from SnackBarThemeData', (WidgetTester tester) async {
+    final SnackBarThemeData snackBarTheme = _snackBarTheme();
+
+    await tester.pumpWidget(MaterialApp(
+      theme: ThemeData(snackBarTheme: snackBarTheme),
+      home: Scaffold(
+        body: Builder(
+            builder: (BuildContext context) {
+              return GestureDetector(
+                onTap: () {
+                  Scaffold.of(context).showSnackBar(SnackBar(
+                    content: const Text('I am a snack bar.'),
+                    duration: const Duration(seconds: 2),
+                    action: SnackBarAction(label: 'ACTION', onPressed: () {}),
+                  ));
+                },
+                child: const Text('X'),
+              );
+            }
+        ),
+      ),
+    ));
+
+    await tester.tap(find.text('X'));
+    await tester.pump(); // start animation
+    await tester.pump(const Duration(milliseconds: 750));
+
+    final Material material = _getSnackBarMaterial(tester);
+    final RawMaterialButton button = _getSnackBarButton(tester);
+
+    expect(material.color, snackBarTheme.backgroundColor);
+    expect(material.elevation, snackBarTheme.elevation);
+    expect(material.shape, snackBarTheme.shape);
+    expect(button.textStyle.color, snackBarTheme.actionTextColor);
```

Comment:
I know it's not easy because it's not a property passed down to a `Material` or something, but ideally you would have an expect in these tests that somehow verifies the `behavior`. 

Maybe you could check the width of the `material`?

Reply:
I added 2 new tests that simply modify the behavior in the theme to either fixed or floating and ensure that the snackbar's position is correct in relation to a FAB. Similar tests exist in `snack_bar_test.dart`